src/server/controllers/changePassword.ts

```diff
@@ -19,204 +19,218 @@ import {
-            });
-          }
-
-          const html = renderer(
-            `${path}/:token`,
-            { requestState, pageTitle },
-            { token },
-          );
-          return res.status(status).type('html').send(html);
-        } catch (error) {
-          logger.error(`${req.method} ${req.originalUrl}  Error`, error);
-          // if theres an error with the token validation, we have to take them back
-          // to the resend page
-          return res.type('html').send(
-            renderer(`${path}/resend`, {
-              requestState,
-              pageTitle: `Resend ${pageTitle} Email`,
-            }),
-          );
-        }
+            ],
+          },
+        });
+      } else {
+        requestState = deepmerge(requestState, {
+          pageData: {
+            email,
+            timeUntilTokenExpiry,
+          },
+          globalMessage: {
+            error: message,
+          },
+        });
       }
+
+      const html = renderer(
+        `${path}/:token`,
+        { requestState, pageTitle },
+        { token },
+      );
+      return res.status(status).type('html').send(html);
+    } catch (error) {
+      logger.error(`${req.method} ${req.originalUrl}  Error`, error);
+      // if theres an error with the token validation, we have to take them back
+      // to the resend page
+      return res.type('html').send(
+        renderer(`${path}/resend`, {
+          requestState,
+          pageTitle: `Resend ${pageTitle} Email`,
+        }),
+      );
     }
-  });
+  }
+};
 
-const OktaSetPassword = async (
+const changePasswordInOkta = async (
   path: PasswordRoutePath,
   pageTitle: PasswordPageTitle,
   req: Request,
   res: ResponseWithRequestState,
+  successRedirectPath: RoutePaths,
 ) => {
   const { stateToken } = readEncryptedStateCookie(req) ?? {};
   const { password } = req.body;
 
   try {
+    validatePasswordFieldForOkta(password);
+
     if (stateToken) {
       const { sessionToken } = await resetPasswordInOkta({
         stateToken,
         newPassword: password,
       });
-      trackMetric('OktaSetPasswordOnWelcomePage::Success');
+
+      trackMetric('OktaUpdatePassword::Success');
 
       if (path === '/welcome') {
         updateEncryptedStateCookie(req, res, {
           passwordSetOnWelcomePage: true,
         });
       }
 
-      // TODO: Authentication in Okta, this kicks off a oauth code flow but will 404 until authentication is built
-      const params = {
-        client_id: okta.clientId,
-        response_type: 'code',
-        scope: 'openid',
-        prompt: 'none',
-        redirect_uri: `https://${baseUri}/oauth/authorization-code/callback`,
-        sessionToken,
-      };
-      return res.redirect(
-        303,
-        `${okta.orgUrl}/oauth2/${okta.authServerId}/v1/authorize?${stringify(
-          params,
-        )}`,
-      );
+      // TODO: once sign out with Okta has been implemented, invalidate current sessions before kicking off code flow
```

Comment:
Moving over the ete tests for Okta should pick this up, as well as some other bits that are missing (passing in `ref` and `refviewid` is also missing as we can't pass them as url parameters when sending emails from Okta). I think this can be left to a future pr, so as not to bloat this one